pkg/controller/job/job_controller.go

```diff
@@ -640,13 +640,11 @@ func (jm *Controller) getPodsForJob(ctx context.Context, j *batch.Job, withFinal
 		return fresh, nil
 	})
 	var finalizers []string
-	if withFinalizers {
-		finalizers = append(finalizers, batch.JobTrackingFinalizer)
-	}
+	finalizers = append(finalizers, batch.JobTrackingFinalizer)
 	cm := controller.NewPodControllerRefManager(jm.podControl, j, selector, controllerKind, canAdoptFunc, finalizers...)
```

Comment:
```suggestion
	cm := controller.NewPodControllerRefManager(jm.podControl, j, selector, controllerKind, canAdoptFunc, batch.JobTrackingFinalizer)
```

Reply:
That doesn't seem right.  What about Pods that have existing finalizers?  Wouldn't this overwrite those?

Reply:
`NewPodControllerRefManager` handles that, IIRC. In any case, my proposal is not changing the logic. It's just reducing the number of lines.

Reply:
So what the previous code is doing is appending the batch.JobTrackingFinalizer to w/e was set in the pods.  I think your proposal would just pass in batch.JobTrackingFinalizer.

So NewPodControllerRefManager:

```
func NewPodControllerRefManager(
	podControl PodControlInterface,
	controller metav1.Object,
	selector labels.Selector,
	controllerKind schema.GroupVersionKind,
	canAdopt func(ctx context.Context) error,
	finalizers ...string,
) *PodControllerRefManager {
	return &PodControllerRefManager{
		BaseControllerRefManager: BaseControllerRefManager{
			Controller:   controller,
			Selector:     selector,
			CanAdoptFunc: canAdopt,
		},
		controllerKind: controllerKind,
		podControl:     podControl,
		finalizers:     finalizers,
	}
}
```
The NewPodControllerRefManager sets the finalizer based on w/e you pass in so I do think this would introduce a bug.

Reply:
The existing code is this:

```golang
	var finalizers []string
	finalizers = append(finalizers, batch.JobTrackingFinalizer)
```

That's not taking into consideration the existing finalizers in the pod template. Those are handled internally, I'm pretty sure.

Reply:
Silly me.  You are correct.